modules/stochastic_tools/src/vectorpostprocessors/StochasticResults.C

```diff
@@ -15,32 +15,106 @@
 
 registerMooseObject("StochasticToolsApp", StochasticResults);
 
-template <>
+defineLegacyParams(StochasticResults);
+
 InputParameters
-validParams<StochasticResults>()
+StochasticResults::validParams()
 {
-  InputParameters params = validParams<GeneralVectorPostprocessor>();
+  InputParameters params = GeneralVectorPostprocessor::validParams();
   params.addClassDescription(
       "Storage container for stochastic simulation results coming from a Postprocessor.");
-  params += validParams<SamplerInterface>();
+  params += SamplerInterface::validParams();
+
+  params.addParam<std::vector<SamplerName>>("samplers",
+                                            "A list of sampler names of associated data.");
+
+  MooseEnum parallel_type("REPLICATED DISTRIBUTED", "REPLICATED");
+  params.addParam<MooseEnum>(
+      "parallel_type",
+      parallel_type,
```

Comment:
If we want to use 'distributed' option, what is the best way to output results from all processors to csv? A single csv file contains all results or each csv file for each rank?

Reply:
See #14480